src/util/adb.js

```diff
@@ -210,10 +210,46 @@ export default class ADBUtils {
     await this.runShellCommand(deviceId, ['mkdir', '-p', artifactsDir]);
 
     this.artifactsDirMap.set(deviceId, artifactsDir);
-
     return artifactsDir;
   }
 
+  async checkOrCleanArtifacts(
+    deviceId: string, remove?: boolean): Promise<boolean> {
+    const {adbClient} = this;
+
+    let directories = false;
```

Comment:
This `directories` is now a boolean, but its name kind of suggests that it is an array (and so the name is misleading).
Also, there is no need to define a state variable outside of the wrapped function, if you look at the definition of wrapADBCall you'll see that it does return a promise that resolves to the value returned by the wrapped function (see https://github.com/mozilla/web-ext/blob/e60411282655cbb23426544b8b5658f435505394/src/util/adb.js#L30) and so we can instead:
- change `await wrapADBCall` to `return wrapADBCall(...)`
- rename `directories` to `found` (as it is easier to recognize it as a boolean) and move it into the wrapped function
- then `return true` in `if (!remove) (because we don't need to look further in that case and we can just return earlier
- then set found to true after the `if (!remove) { return true; }` and return the value set to `found` at the end on the wrapped function